src/lib.rs

```diff
@@ -92,76 +126,23 @@ pub fn hash(data: &[u8]) -> [u8; 32] {
     hashing_impl::sha3_256(data)
 }
 
-/// Represents a set of public keys, consisting of a public signature key and a public
-/// encryption key.
+/// Use `gen_encrypt_keypair()` to generate a public and private key pair.
 #[derive(Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, Clone)]
-pub struct PublicKeys {
-    sign: sign::PublicKey,
+pub struct PublicEncryptKey {
     encrypt: box_::PublicKey,
 }
 
-/// Secret counterpart of the public key set, consisting of a secret signing key and
-/// a secret encryption key.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct SecretKeys {
-    inner: Arc<SecretKeysInner>,
-    public: PublicKeys,
-}
-
-#[derive(Debug, PartialEq, Eq)]
-struct SecretKeysInner {
-    sign: sign::SecretKey,
-    encrypt: box_::SecretKey,
-}
-
-/// Secret key for authenticated symmetric encryption.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct SymmetricKey {
-    encrypt: Arc<secretbox::Key>,
-}
-
-/// Detached signature.
-#[derive(Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, Clone)]
-pub struct Signature {
-    signature: sign::Signature,
-}
-
-/// Precomputed shared secret key.
-/// Can be created from a pair of our secret key and the recipient's public key.
-/// As a result, we'll get the same key as the recipient with their secret key and
-/// our public key.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct SharedSecretKey {
-    precomputed: Arc<box_::PrecomputedKey>,
-}
-
-#[derive(Serialize, Deserialize)]
-struct CipherText {
-    nonce: [u8; box_::NONCEBYTES],
-    ciphertext: Vec<u8>,
-}
+impl PublicEncryptKey {
+    /// Construct public key from bytes. Useful when it was serialized before.
```

Comment:
We try to stick to UK spellings rather than US wherever possible, so "serialised" rather than "serialized".